clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -500,8 +500,9 @@ public int read(ByteBuffer dst) throws IOException {
             read = readFromAppBuffer(dst);
         }
 
-        int netread = 0;
-        if (dst.remaining() > 0) {
+        boolean isClosed = false;
+        while (dst.remaining() > 0) {
```

Comment:
A short comment explaining the need for this loop would be helpful.

Reply:
@hachikuji Good point. Thanks much for taking time to review the patch.

I have updated the patch to include the following comment. Not sure if it is good enough:

```
// Keep reading from the socket into the dst buffer until either the dst buffer is full or there is no more
// data in the socket. Each loop read at most once from the socket.
```